// Proto file representing the LCSLayer class

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";
import "layer.proto";

// Stackup layer service definition
service StackupLayerService {
  // Creates a stackup layer
  rpc Create (StackupLayerCreationMessage) returns (EDBObjMessage) {}

  // Set the negative property of the stackup layer
  rpc SetNegative (SetNegativeMessage) returns (google.protobuf.Empty) {}
}

// Message for StackupLayerService::Create
message StackupLayerCreationMessage {
  string name = 1;
  LayerType type = 2;
  ValueMessage thickness = 3;
  ValueMessage elevation = 4;
  string material = 5;
}

// Message for StackupLayerService::SetNegative
message SetNegativeMessage {
  EDBObjMessage layer = 1;
  bool is_negative = 2;
}
